src/db/pool.cr

```diff
@@ -216,11 +246,27 @@ module DB
 
     private def build_resource : T
       resource = @factory.call
+      resource.connect!(self, @total.size) if resource.responds_to?(:connect!)
       @total << resource
       @idle << resource
       resource
     end
 
+    def replace_expired_resource!(expired_resource : T)
+      reaped = false
+
+      each_resource do |cnn|
```

Comment:
It took me a few reads to understand that cnn is a connection

Reply:
Yeah, they use `cnn` internally a lot so I stuck with it, but I don't love it necessary either